### PR TITLE
OL-2559 - Adding/updating README file

### PR DESCRIPTION
**Pull request by narayanangp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,184 @@
 # NSI_OL052_TLS02_SEND_EMAIL
-This is the **service** to send email
+The **email send service** is used to send emails of all types and to send live Triggered Email(s).
```
Comment (pkaliniak) [inline README.md]: not really of all types _and_ live triggered emails. Please take look at summary from swagger. We updated that according to Mike's comment.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,184 @@
...
+```bash
+docker inspect -f '{{.NetworkSettings.Networks.fuse.IPAddress}}' ol052-tls02-send-email
+```
+
+### Case examples:
+1) Try to send an email with a valid request
+
+Request:
+```bash
+curl -v -XPOST -H "apiRequestOriginator:123" -H "RequestID:6948DF80-14BD-4E04-8842-7668D9C001F5" "Content-Type: application/json" -d 'http://127.0.0.1:1080/api/json/email_send/method=triggerFolder&folderId=47&emailAddress=matthew.2.johnson@atos.net&profileData={"TFC":{"RecipientName":"Matthew","ProductShortName":"Childcareservice","ProductName":"Childcare service"}}'
```
Comment (pkaliniak) [inline README.md]: * wrong path, should be `/messanging/v1/emails` or smth like that - please check swagger
* wrong way of passing data. This is `applicaiton/json` while the way you try to pass it is url-encoded.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,184 @@
...
+``````
+
+Response:
+Http code 204
+
+2) Try to send an email with the no body
+
+Request:
+```bash
+curl -v -XPOST -H "apiRequestOriginator:123" -H "RequestID:6948DF80-14BD-4E04-8842-7668D9C001F5" "Content-Type: application/application/json" -d 'http://127.0.0.1:1080/api/json/email_send'
```
Comment (pkaliniak) [inline README.md]: wrong path

File: README.md [hunk 4]
```diff
@@ -1,2 +1,184 @@
...
+    "errorCode": "E0000"
+    "message": "Invalid input data"
+}
+```
+
+3) Try to send an email, when 'apiRequestOriginator' header is missing
+
+Request:
+```bash
+curl -v -XPOST -H "RequestID:6948DF80-14BD-4E04-8842-7668D9C001F5" "Content-Type: application/application/json" -d 'http://127.0.0.1:1080/api/json/email_send/method=triggerFolder&folderId=47&emailAddress=matthew.2.johnson@atos.net&profileData={"TFC":{"RecipientName":"Matthew","ProductShortName":"Childcareservice","ProductName":"Childcare service"}}'
```
Comment (pkaliniak) [inline README.md]: wrong path and way of passing data

File: README.md [hunk 5]
```diff
@@ -1,2 +1,184 @@
...
+    "errorCode": "E0001"
+    "message": "apiRequestOriginator is missing"
+}
+```
+
+4) Try to send an email, when 'RequestID' header is missing
+
+Request:
+```bash
+curl -v -XPOST -H "apiRequestOriginator:123" "Content-Type: application/application/json" -d 'http://127.0.0.1:1080/api/json/email_send/method=triggerFolder&folderId=47&emailAddress=matthew.2.johnson@atos.net&profileData={"TFC":{"RecipientName":"Matthew","ProductShortName":"Childcareservice","ProductName":"Childcare service"}}'
```
Comment (pkaliniak) [inline README.md]: as above

File: README.md [hunk 6]
```diff
@@ -1,2 +1,184 @@
...
+}
+```
+
+5) Try to send email, when returning 403 error during accessing the Service Layer
+
+The trick for test, this call is to have active service and stub but with no authentication
+
+Request:
+```bash
+curl -v -XPOST -H "apiRequestOriginator:123" -H "RequestID:6948DF80-14BD-4E04-8842-7668D9C001F5" "Content-Type: application/application/json" -d 'http://127.0.0.1:1080/api/json/email_send/method=triggerFolder&folderId=47&emailAddress=matthew.2.johnson@atos.net&profileData={"TFC":{"RecipientName":"Matthew","ProductShortName":"Childcareservice","ProductName":"Childcare service"}}'
```
Comment (pkaliniak) [inline README.md]: as above

File: README.md [hunk 7]
```diff
@@ -1,2 +1,184 @@
...
+}
+```
+
+6) Try to send email, when returning 503 error during accessing the Service Layer
+
+The trick for test, this call is to have active service, but deactivated stub (so the call from service would not be able to connect with stub)
+
+Request:
+```bash
+curl -v -XPOST -H "apiRequestOriginator:123" -H "RequestID:6948DF80-14BD-4E04-8842-7668D9C001F5" "Content-Type: application/application/json" -d 'http://127.0.0.1:1080/api/json/email_send/method=triggerFolder&folderId=47&emailAddress=matthew.2.johnson@atos.net&profileData={"TFC":{"RecipientName":"Matthew","ProductShortName":"Childcareservice","ProductName":"Childcare service"}}'
```
Comment (pkaliniak) [inline README.md]: as above

File: README.md [hunk 8]
```diff
@@ -1,2 +1,184 @@
...
+}
+```
+
+7) Try to send email, when returning 503 error during accessing the Service Layer
+
+The trick for test, this call is to have active service, but stub should not respond (due to technical issue) even though stub is up and running
+
+Request:
+```bash
+curl -v -XPOST -H "apiRequestOriginator:123" -H "RequestID:6948DF80-14BD-4E04-8842-7668D9C001F5" "Content-Type: application/application/json" -d 'http://127.0.0.1:1080/api/json/email_send/method=triggerFolder&folderId=47&emailAddress=matthew.2.johnson@atos.net&profileData={"TFC":{"RecipientName":"Matthew","ProductShortName":"Childcareservice","ProductName":"Childcare service"}}'
```
Comment (pkaliniak) [inline README.md]: as above

File: README.md [hunk 9]
```diff
@@ -1,2 +1,184 @@
...
+    "errorCode": "E8001",
+    "errorDescription": "Service not available due to lack of connection to provider"
+}
+```
+
+8) Simulate receiving Service Layer Internal Server Error
+
+Request:
+```bash
+curl -v -XPOST -H "apiRequestOriginator:123" -H "RequestID:6948DF80-14BD-4E04-8842-7668D9C001F5" "Content-Type: application/application/json" -d 'http://127.0.0.1:1080/api/json/email_send/method=triggerFolder&folderId=47&emailAddress=matthew.2.johnson@atos.net&profileData={"TFC":{"RecipientName":"Matthew","ProductShortName":"Childcareservice","ProductName":"Childcare service"}}'
```
Comment (pkaliniak) [inline README.md]: as above

File: README.md [hunk 10]
```diff
@@ -1,2 +1,184 @@
...
+    "errorCode": "E9000",
+    "errorDescription": "Internal server error"
+}
+```
+
+9) Simulate receiving Service Layer Internal Server Error, due to the provider returns an error condition that was not mapped in the table
+
+Request:
+```bash
+curl -v -XPOST -H "apiRequestOriginator:123" -H "RequestID:6948DF80-14BD-4E04-8842-7668D9C001F5" "Content-Type: application/application/json" -d 'http://127.0.0.1:1080/api/json/email_send/method=triggerFolder&folderId=47&emailAddress=matthew.2.johnson@atos.net&profileData={"TFC":{"RecipientName":"Matthew","ProductShortName":"Childcareservice","ProductName":"Childcare service"}}'
```
Comment (pkaliniak) [inline README.md]: as above

File: README.md (hunk 1 above)
Comment (narayanangp) [reply README.md]: Fixed

**Pull request merged**